test/vm-download

```diff
@@ -68,20 +68,27 @@ def download(link, force, stores):
             continue
 
     sys.stderr.write("{0}\n".format(source))
-    (fd, temp) = tempfile.mkstemp(suffix=".partial", prefix=os.path.basename(dest), dir=DATA)
-    curl = subprocess.Popen(["curl", "-#", "-f", source], stdout=subprocess.PIPE)
-    unxz = subprocess.Popen(["unxz", "--stdout", "-"], stdin=curl.stdout, stdout=fd)
-
-    curl.stdout.close()
-    ret = curl.wait()
-    if ret != 0:
-        sys.exit(ret)
-    ret = unxz.wait()
-    if ret != 0:
-        sys.exit(ret)
-
-    os.close(fd)
-    shutil.move(temp, dest)
+    # use pid in the temp file in case we have multiple downloaders
+    # make sure that it ends with .partial so the prune logic works
+    (fd, temp) = tempfile.mkstemp(suffix=".{0}.partial".format(os.getpid()), prefix=os.path.basename(dest), dir=DATA)
```

Comment:
> The point of using mkstemp is that you should get a unique name each time it's called.

Right. I removed that commit. I misread the mkstemp and thought that we could use the pid info to avoid conflicts, but I realized that the download processes will probably be running on different systems, so info on pid is neither unique nor particularly useful.